File: android/src/main/kotlin/com/kkaun/video_multishare/VideoMultisharePlugin.kt
package com.kkaun.video_multishare

import android.app.Activity
import android.content.Context
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.embedding.engine.plugins.activity.ActivityAware
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class VideoMultisharePlugin : FlutterPlugin, MethodChannel.MethodCallHandler, ActivityAware {

    private lateinit var channel: MethodChannel
    private lateinit var context: Context
    private var activity: Activity? = null

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        context = flutterPluginBinding.applicationContext
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "com.kkaun.video_multi_share")
        channel.setMethodCallHandler(this)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }

    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
        activity = binding.activity
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        val videoPath = call.argument<String>("videoPath")
        if (videoPath.isNullOrEmpty()) {
            result.error("INVALID_ARGUMENT", "Video path is null or empty", null)
            return
        }
        if (activity == null) {
            result.error("INVALID_ARGUMENT", "Attached Activity is null!", null)
            return
        }

        val (isSuccess, errorMessage) = when (call.method) {
            "shareToInstagramStory" -> VideoSharingDelegate.shareToInstagramStory(activity!!, videoPath)
            "shareToInstagramReels" -> VideoSharingDelegate.shareToInstagramWMethodPick(activity!!, videoPath)
            "shareToInstagramDirect" -> VideoSharingDelegate.shareToInstagramWMethodPick(activity!!, videoPath)
            "shareToFacebookPost" -> VideoSharingDelegate.shareToFacebookGenericOptions(activity!!, videoPath)
            "shareToFacebookStory" -> VideoSharingDelegate.shareToFacebookStory(activity!!, videoPath)
            "shareToYouTubeShorts" -> VideoSharingDelegate.shareToYouTubeShorts(activity!!, videoPath)
            "shareToTikTok" -> VideoSharingDelegate.shareToTikTok(activity!!, videoPath)
            "shareToWhatsApp" -> VideoSharingDelegate.shareToWhatsApp(activity!!, videoPath)
            "shareToSnapchat" -> VideoSharingDelegate.shareToSnapchat(activity!!, videoPath)
            else -> {
                result.notImplemented()
                return
            }
        }

        if (isSuccess) {
            result.success(true)
        } else {
            result.error("SHARE_FAILED", errorMessage, null)
        }
    }

    override fun onDetachedFromActivity() {
        activity = null
    }

    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
        activity = binding.activity
    }

    override fun onDetachedFromActivityForConfigChanges() {
        activity = null
    }
}

File: ios/Classes/VideoMultisharePlugin.swift
import Flutter
import UIKit

public class VideoMultisharePlugin: NSObject, FlutterPlugin {
    
    // Required Flutter plugin registration method
    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(name: "com.kkaun.video_multi_share", binaryMessenger: registrar.messenger())
        let instance = VideoMultisharePlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }
    
    // Handle method calls from Flutter
    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        guard let args = call.arguments as? [String: Any],
              let videoPath = args["videoPath"] as? String,
              let videoURL = URL(string: videoPath) else {
            result(FlutterError(code: "INVALID_ARGUMENT", message: "Invalid video path", details: nil))
            return
        }
        
        let viewController = UIApplication.shared.windows.first!.rootViewController!
        
        let (success, errorMessage): (Bool, String) = {
            switch call.method {
            case "shareToInstagramStory":
                return VideoSharingDelegate.shareToInstagramStory(videoURL: videoURL)
            case "shareToInstagramReels":
                return VideoSharingDelegate.shareToInstagramGeneric(videoURL: videoURL, viewController: viewController)
            case "shareToInstagramDirect":
                return VideoSharingDelegate.shareToInstagramGeneric(videoURL: videoURL, viewController: viewController)
            case "shareToFacebookPost":
                return VideoSharingDelegate.shareToFacebookGeneric(videoURL: videoURL, viewController: viewController)
            case "shareToFacebookStory":
                return VideoSharingDelegate.shareToFacebookStory(videoURL: videoURL)
            case "shareToYouTubeShorts":
                return VideoSharingDelegate.shareToYouTube(videoURL: videoURL)
            case "shareToWhatsApp":
                return VideoSharingDelegate.shareToWhatsApp(videoURL: videoURL)
            case "shareToTikTok":
                guard let redirectURI = args["redirectURI"] as? String else {
                    result(FlutterError(code: "INVALID_ARGUMENT", message: "Missing redirectURI", details: nil))
                    return (false, "Missing redirectURI")
                }
                VideoSharingDelegate.shareToTikTok(videoURL: videoURL, redirectURI: redirectURI) { shareResult in
                    DispatchQueue.main.async {
                        switch shareResult {
                        case .success(let message):
                            result(message)
                        case .failure(let error):
                            result(FlutterError(code: "SHARE_FAILED", message: error.localizedDescription, details: nil))
                        }
                    }
                }
            case "shareToSnapchat":
                VideoSharingDelegate.shareToSnapchat(videoURL: videoURL, viewController: viewController) { shareResult in
                    switch shareResult {
                    case .success(let message):
                        result(message)
                    case .failure(let error):
                        result(FlutterError(code: "SHARE_FAILED", message: error.localizedDescription, details: nil))
                    }
                }
            default:
                return (false, "Method not implemented")
            }
            return (false, "Method not implemented")
        }()
        
        if success {
            result(true)
        } else {
            result(FlutterError(code: "SHARE_FAILED", message: errorMessage, details: nil))
        }
    }
}
